acme/acme/crypto_util.py

```diff
@@ -280,3 +282,23 @@ def gen_ss_cert(key, domains, not_before=None,
     cert.set_pubkey(key)
     cert.sign(key, "sha256")
     return cert
+
+def dump_pyopenssl_chain(chain, filetype=OpenSSL.crypto.FILETYPE_PEM):
```

Comment:
We may want to consider modifying this function before the release or making it private, but I don't care too much. The reason for this is it forces the user of the module to use `OpenSSL` (which is slowly going away in favor of `cryptography`) and it returns bytes instead of strings.

Reply:
It's like that because `fetch_chain` returns `OpenSSL` objects, which isn't going to change.

Reply:
> The reason for this is it forces the user of the module to use OpenSSL (which is slowly going away in favor of cryptography)

While this applied to `chain` as you pointed out, we could remove the need to use it for `filetype`, but I think it is a pretty minor point and making separate functions or types to hide this from the user doesn't buy us much and makes our code more complicated.

> it returns bytes instead of strings.

I'm confident this is correct for a function like this. See https://github.com/certbot/certbot/issues/5606#issuecomment-369768427.